### ISSUE TITLE
Explain to me like I'm 5, what the pros/con are each data_structure

### ISSUE THREAD

**Issue by BaReinhard** [BaReinhard/Hacktoberfest-Data-Structure-and-Algorithms#133]
Create a file in the descriptions folder, choose one data_structure or algorithm and give a brief explanation of what the pros and cons are (like I'm 5). Then explain in deep details the pros and cons of the structure, giving the best descriptions possible.

Files  Structure:
```
/descriptions/data_structure.md
ex: /descriptions/breadth_first_search.md
```

File Content Structure:
```
## ELI5
Some text explaining the algorithm to someone who is 5 aka not too technical, but should explain enough
### Pros
* Pros
### Cons
* Cons

## Technical Explanation
Some text explaining in greater detail.
### Pros
* Pros
### Cons
* Cons
```

**Comment by mhdsmr2** [comment]
May I work on it?

**Comment by mhdsmr2** [comment]
Let's just wait for @BaReinhard to be online.

**Comment by BaReinhard** [comment]
Anyone can work on it. The length isn't important but it should be as detailed as possible to gain a full understanding of the algorithm or data structure.

**Comment by mhdsmr2** [comment]
When will you close this issue? Would you close it after I PR a 6 algorithm/data structure description? Because, as far as I know, Hacktober counts a PR as a contribution if it closes an issue (tell me if I am wrong).

**Comment by okatticus** [comment]
@BaReinhard I'm interested in working on it. 
@mhdsmr2  Let us divide the data structures and finish all descriptions.

**Comment by mhdsmr2** [comment]
So how we gonna divide it?

**Comment by okatticus** [comment]
@mhdsmr2  I'll take all the data structures and first five algorithms and you take the rest of the algorithms.
 Works fine?

**Comment by mhdsmr2** [comment]
And where is the list of Data Structures and Algos you're talking about?

**Comment by mhdsmr2** [comment]
Yeah, great!

**Comment by okatticus** [comment]
In the repository
Algos:
https://github.com/BaReinhard/Hacktoberfest-Data-Structure-and-Algorithms/tree/master/algorithms
Data Structures: 
https://github.com/BaReinhard/Hacktoberfest-Data-Structure-and-Algorithms/tree/master/data_structures

**Comment by mhdsmr2** [comment]
Yea. I got it.

**Comment by mhdsmr2** [comment]
Hey, @BaReinhard, Can you have a look at my forked repo and see the explanation of Binary Search and tell me if I did explain it well enough.

**Comment by okatticus** [comment]
#158

**Comment by mhdsmr2** [comment]
@okatticus I think you should delete your last comment because it's already showing that you referred to this issue in your PR. Thanks if you did delete your comment.

**Comment by brilam** [comment]
@BaReinhard Could you review my work for HashTable and see if it is okay to merge? :)

**Comment by Bahukhandi93** [comment]
please assign this issue to me. Thank you!
